File: dynamicProgramming/memoizingFibo.cpp
#include <iostream>
#include <map>

int mfib(int num)
{
    static std::map<int, int> memo{ { 0, 0 }, { 1, 1 } };

    auto found = memo.find(num);
    if (found != std::end(memo)) {
        return found->second;
    }

    int result = mfib(num - 1) + mfib(num - 2);
    memo[num] = result;
    return result;
}

int main(void)
{
    std::cout << mfib(8) << std::endl;
    std::cout << mfib(10000) << std::endl;
}

File: dynamicProgramming/memoizingFibo.js
var memoize = function(f) {
  var memo = [];
  return function(i) {
    if (memo[i]) {
      return memo[i];
    }
    result = f(i);
    memo[i] = result;
    return result;
  }
}

var mfib = memoize(function(i) {
  if (i === 0 || i === 1) {
    return i;
  }
  return mfib(i - 1) + mfib(i - 2);
})

console.log(mfib(8));
console.log(mfib(1000));
